pallets/amm/src/tests.rs

```diff
@@ -283,6 +283,149 @@ fn remove_zero_liquidity_should_not_work() {
 	});
 }
 
+#[test]
+fn remove_liquidity_should_not_work() {
+	new_test_ext().execute_with(|| {
+		let user = ALICE;
+		let asset_a = HDX;
+		let asset_b = DOT;
+
+		assert_ok!(AMM::create_pool(
+			Origin::signed(user),
+			asset_a,
+			asset_b,
+			100_000_000,
+			Price::from(10_000)
+		));
+
+		let asset_pair = AssetPair {
+			asset_in: asset_a,
+			asset_out: asset_b,
+		};
+		assert_eq!(AMM::exists(asset_pair), true);
+
+		assert_noop!(
+			AMM::remove_liquidity(Origin::signed(user), asset_a, asset_b, 100_000_000 - MIN_POOL_LIQUIDITY_LIMIT + 1),
+		 	Error::<Test>::MinimalPoolLiquidityRequirementNotMet
+		);
+	});
+}
+
+#[test]
+fn remove_liquidity_should_not_lock_user() {
+	new_test_ext().execute_with(|| {
+		let user_one = ALICE;
+		let user_two = BOB;
+		let asset_a = HDX;
+		let asset_b = DOT;
+
+		assert_ok!(AMM::create_pool(
+			Origin::signed(user_one),
+			asset_a,
+			asset_b,
+			100_000_000,
+			Price::from(10_000)
+		));
+
+		let asset_pair = AssetPair {
+			asset_in: asset_a,
+			asset_out: asset_b,
+		};
+		assert_eq!(AMM::exists(asset_pair), true);
```

Comment:
same here. assert would do it

```suggestion
		assert!(AMM::exists(asset_pair));
```